spinalcordtoolbox/scripts/sct_apply_transfo.py

```diff
@@ -109,6 +111,43 @@ def get_parser():
     return parser
 
 
+def isct_antsApplyTransforms(dimensionality,
+                             fname_input,
+                             fname_output,
+                             fname_reference,
+                             transforms,
+                             interpolation_args,
+                             verbose):
+    """
+    Wrapper for CLI binary that allows for pre/postprocessing steps.
+
+    Note: Most logic should still go in `sct_apply_transfo`, because our SCT
+    script is essentially one big wrapper script for the ANTs call. Only use
+    this wrapper to fix bugs in the binary itself, or to compensate for
+    missing behavior/options in the ANTs source code. This wrapper should more
+    or less function like the original binary (with only minor tweaks added).
+    """
+    run_proc(['isct_antsApplyTransforms',
+              '-d', dimensionality,
+              '-i', fname_input,
+              '-o', fname_output,
+              '-t'] + transforms +  # list of transforms
+             ['-r', fname_reference] +
+             interpolation_args,  # list of ['-n', interp_type]
+             verbose=verbose,
+             is_sct_binary=True)
+    # Preserve integer datatype when interpolation is NearestNeighour to
+    # counter ANTs behavior of always outputting float64 images
+    im_out = Image(fname_output)
+    dtype_in = Image(fname_input).data.dtype
+    input_is_int = issubclass(dtype_in.type, np.integer)
+    if input_is_int and 'NearestNeighbor' in interpolation_args:
```

Comment:
It might be even simpler to just have the rule that NearestNeighbor interpolation outputs the same dtype as the input, regardless of integer/float/etc.? But I'm happy either way.
```suggestion
    if 'NearestNeighbor' in interpolation_args:
```